herd/ASLAction.ml

```diff
@@ -90,65 +114,97 @@ module Make (A : S) = struct
   (* relative to memory *)
   let is_mem_store = function
     | Access (W, A.Location_global _, _, _, _) -> true
-    | _ -> false
+    | Access _| Branching _|Barrier _|TooFar _|NoAction
+      -> false
 
   let is_mem_load = function
     | Access (R, A.Location_global _, _, _, _) -> true
-    | _ -> false
+    | Access _| Branching _|Barrier _|TooFar _|NoAction
+      -> false
 
-  let is_additional_mem_load _action = false
+  let to_something r = function
+    | Access _| Branching _|Barrier _|TooFar _|NoAction
+      -> r
+
+  let to_false = to_something false
+  let to_true = to_something true
+  let to_none = to_something None
```

Comment:
I'd rather not follow you here. The strange coding with an apparently useless pattern-matching expression will trigger a compiler warning in case we add some constructor.  I notice that this warning is not ideal as the potential problem is likely to occur more at `to_false`, `to_true`, etc. usage than definition. However this is far better than no warning at all...

Reply:
We should even do more : inline `to_something`. I'll do that.

Reply:
I mean inline all four functions.

Reply:
I did so (just pushed)